azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/sync_clients.py

```diff
@@ -165,14 +188,28 @@ def __init__(self, transport):
         :param transport: The transport that the client will use.
         """
         super(DeviceClient, self).__init__(transport)
-        self._transport.on_transport_c2d_message_received = self._queue_manager.route_c2d_message
+        self._transport.on_transport_c2d_message_received = self._inbox_manager.route_c2d_message
 
-    def get_c2d_message_queue(self):
-        """Returns a MessageQueue object that can be used to receive "Cloud to Device" messages.
+    def receive_c2d_message(self, block=True, timeout=None):
+        """Receive a C2D message that has been sent from the Azure IoT Hub.
 
-        :returns: MessageQueue object for C2D Messages.
+        :param bool block: Indicates if the operation should block until a message is received.
+        Default True.
+        :param int timeout: Optionally provide a number of seconds until blocking times out.
+
+        :raises: InboxEmpty if timeout occurs on a blocking operation.
+        :raises: InboxEmpty if no message is available on a non-blocking operation.
```

Comment:
We can change this later, but I would think we'd want a timeout exception on timeout.  For no message on non-blocking op, maybe we want a "no message" exception.  exposing an exception with the name "Inbox" feels like it's leaking an implementation detail.

Reply:
Yeah, I also agree, but I think this will require some kind of formalization of the error story.

---
In reply to: [264339874](https://github.com/Azure/azure-iot-sdk-python-preview/pull/55#discussion_r264339874) [](ancestors = 264339874)